python/cuml/dask/ensemble/randomforestregressor.py

```diff
@@ -277,17 +275,20 @@ def _func_build_rf(
         )
 
     @staticmethod
-    def _fit(model, X_df_list, y_df_list, r):
-        if len(X_df_list) != len(y_df_list):
-            raise ValueError("X (%d) and y (%d) partition list sizes unequal" %
-                             len(X_df_list), len(y_df_list))
-        if len(X_df_list) == 1:
-            X_df = X_df_list[0]
-            y_df = y_df_list[0]
+    def _fit(model, input_data, convert_dtype):
+        X = input_data[0][0]
+        y = input_data[0][1]
+        if isinstance(X, cudf.core.DataFrame):
+            if len(input_data) > 1:
+                for i in range(1, len(input_data)):
+                    X = cudf.concat([X, input_data[i][0]])
```

Comment:
why run multiple concats here? would just one concat be faster and simpler like: `X = cudf.concat([item[0] for item in input_data])`

Reply:
actually, see comment in the other `_fit` - I think input_utils.concatenate + list comprehension will really reduce the code here.